sdk/client/src/response.rs

```diff
@@ -0,0 +1,141 @@
+        let response = match method {
+            Method::Submit => MethodResponse::Submit,
+            Method::GetMetadata => MethodResponse::GetMetadata(serde_json::from_value(json)?),
+            Method::GetAccount => MethodResponse::GetAccount(serde_json::from_value(json)?),
+            Method::GetTransactions => {
+                MethodResponse::GetTransactions(serde_json::from_value(json)?)
+            }
+            Method::GetAccountTransaction => {
+                MethodResponse::GetAccountTransaction(serde_json::from_value(json)?)
+            }
+            Method::GetAccountTransactions => {
+                MethodResponse::GetAccountTransactions(serde_json::from_value(json)?)
+            }
+            Method::GetEvents => MethodResponse::GetEvents(serde_json::from_value(json)?),
+            Method::GetCurrencies => MethodResponse::GetCurrencies(serde_json::from_value(json)?),
+            Method::GetNetworkStatus => {
+                MethodResponse::GetNetworkStatus(serde_json::from_value(json)?)
+            }
+            Method::GetStateProof => MethodResponse::GetStateProof(serde_json::from_value(json)?),
+            Method::GetAccountStateWithProof => {
+                MethodResponse::GetAccountStateWithProof(serde_json::from_value(json)?)
+            }
+            Method::GetTransactionsWithProofs => MethodResponse::GetTransactionsWithProofs,
+            Method::GetEventsWithProofs => MethodResponse::GetEventsWithProofs,
+        };
+
+        Ok(response)
+    }
+
+    pub fn method(&self) -> Method {
+        match self {
+            MethodResponse::Submit => Method::Submit,
+            MethodResponse::GetMetadata(_) => Method::GetMetadata,
+            MethodResponse::GetAccount(_) => Method::GetAccount,
+            MethodResponse::GetTransactions(_) => Method::GetTransactions,
+            MethodResponse::GetAccountTransaction(_) => Method::GetAccountTransaction,
+            MethodResponse::GetAccountTransactions(_) => Method::GetAccountTransactions,
+            MethodResponse::GetEvents(_) => Method::GetEvents,
+            MethodResponse::GetCurrencies(_) => Method::GetCurrencies,
+            MethodResponse::GetNetworkStatus(_) => Method::GetNetworkStatus,
+            MethodResponse::GetStateProof(_) => Method::GetStateProof,
+            MethodResponse::GetAccountStateWithProof(_) => Method::GetAccountStateWithProof,
+            MethodResponse::GetTransactionsWithProofs => Method::GetTransactionsWithProofs,
+            MethodResponse::GetEventsWithProofs => Method::GetEventsWithProofs,
+        }
+    }
+
+    pub fn unwrap_get_account(self) -> Option<AccountView> {
+        if let MethodResponse::GetAccount(inner) = self {
+            inner
+        } else {
+            panic!(
+                "expected MethodResponse::GetAccount found MethodResponse::{:?}",
+                self.method()
+            );
+        }
+    }
+}
```

Comment:
Curious, wouldn't you want this for basically all the `MethodResponse?`. Maybe a macro for it and you can use it for all of them?

Reply:
True, this was the only one that was currently needed to remove some boilerplate. I'll look into your suggestion for the future.